nautobot/extras/models/relationships.py

```diff
@@ -736,3 +774,65 @@ def _validate_relationship_filter_restriction(self):
                 raise ValidationError(
                     {side_name: (f"{side} violates {self.relationship} {side_name}_filter restriction")}
                 )
+
+
+def get_relationships_errors(request, obj, output_for="ui"):
```

Comment:
So this doesn't actually check for relationship errors - it only checks for the case where none of the required objects exist and tells the user to create them first? I wonder if something like `check_for_missing_required_related_objects` might be a clearer name.

I also wonder if it'd work any better for this to be a class method or instance method on the `RelationshipModel` class instead of a standalone function?